TundraTD/Assets/Scripts/ModulesUI/MagicScreen/PlayerDeck.cs

```diff
@@ -10,20 +12,39 @@ namespace ModulesUI.MagicScreen
     /// </summary>
     public static class PlayerDeck
     {
+        public static BasicElement CurrentMostElement;
         public static ObservableCollection<BasicElement> DeckElements { get; } = new ObservableCollection<BasicElement>();
+        public static Dictionary<BasicElement, int> ElementsQuantity { get; } 
         public static Dictionary<BasicElement, Sprite> ElementIcons { get; }
 
         static PlayerDeck()
         {
-            var loadedIcons = Resources.LoadAll<Sprite>("Elements/");
+            var loadedIcons = new Sprite[6];
+            int i = 0;
+            foreach (var element in (BasicElement[])Enum.GetValues(typeof(BasicElement)))
+            {
+                loadedIcons[i] = Resources.Load<Sprite>($"Elements/{element}");
+                i++;
+            }
+            //var loadedIcons = Resources.LoadAll<Sprite>("Elements/");
             ElementIcons = new Dictionary<BasicElement, Sprite>
             {
-                [BasicElement.Air] = loadedIcons[1],
-                [BasicElement.Earth] = loadedIcons[2],
-                [BasicElement.Fire] = loadedIcons[3],
+                [BasicElement.None] = loadedIcons[0],
+                [BasicElement.Fire] = loadedIcons[1],
+                [BasicElement.Water] = loadedIcons[2],
+                [BasicElement.Earth] = loadedIcons[3],
                 [BasicElement.Lightning] = loadedIcons[4],
-                [BasicElement.Water] = loadedIcons[5],
-                [BasicElement.None] = null
+                [BasicElement.Air] = loadedIcons[5]
+            };
+
+            ElementsQuantity = new Dictionary<BasicElement, int>
```

Comment:
I think keeping these variables is actually not the best solution. Keeping them in list and drawing in this order was better because it's simple to understand and get data and also it requires less memory. To get elements with their amount, you can use the `Enumerable.GroupBy(Func<> selector)` method in `System.Linq` namespace. It returns grouped enumerable which contains objects of `IGrouping` type: they contain the `Key` property which represents the grouping parameter (in our case it is element) and IEnumerable of elements, `Count()` of which you can get to count elements in group

Reply:
Not quiet sure how to use GroupBy but implemented counting and highlighting feature in a4160e2 without counting dictionary